minimum_jerk_ros/src/minimum_jerk_ros.cpp

```diff
@@ -0,0 +1,736 @@
+        double dist = abs(pose_target.get_theta() - pose_start.get_theta());
+        double max_total = dist / goal->min_rotational_vel;
+        TrajectoryPlanner controller = TrajectoryPlanner(max_total, 1 / this->control_frequency_);
+        Robot robot = Robot("Robot", max_total, controller, pose_start, pose_target, "r");
+        robot.generate_trajectory();
+
+        auto odometry = robot.get_odometry();
+
+        if (goal->enable_data_save)
+        {
+            FilFile::fil_file_rotation<Pose>(robot.get_odometry().get_timestamps(), robot.get_odometry().get_poses(), "./src/mypackage/minimum_jerk_relative_move/minimum_jerk_trajectory_planner/ressources/poses.txt");
+            FilFile::fil_file_rotation<Velocity>(robot.get_odometry().get_timestamps(), robot.get_odometry().get_velocities(), "./src/mypackage/minimum_jerk_relative_move/minimum_jerk_trajectory_planner/ressources/velocities.txt");
+            FilFile::fil_file_rotation<Acceleration>(robot.get_odometry().get_timestamps(), robot.get_odometry().get_accelerations(), "./src/mypackage/minimum_jerk_relative_move/minimum_jerk_trajectory_planner/ressources/accelerations.txt");
+        }
+        return odometry;
+    }
+
+    bool MinimumJerkRos::check_rotation_collision_(std::vector<geometry_msgs::msg::PoseStamped> poses, geometry_msgs::msg::Twist angular_vel)
+    {
+        auto start = std::chrono::system_clock::now();
+        // If there's an obstacle between the current pose and the horizon distance on the pre-planned path, stop the robot
+        // The robot stops for idle_timeout_ seconds maximum
+        // If the path wasn't cleared in that time, the action fails, recovery is to be handled in the behavior trees
+
+        while (is_obstacle_on_path_(poses))
+        {
+            if (action_rotation_server_->is_cancel_requested())
+            {
+                RCLCPP_WARN(get_logger(), "Goal was canceled. Canceling.");
+                stop_rotation_(angular_vel);
+                action_rotation_server_->terminate_all();
+                return false;
+            }
+
+            stop_rotation_(angular_vel);
+            auto end = std::chrono::system_clock::now();
+            std::chrono::duration<double> elapsed_seconds = end - start;
+
+            if (elapsed_seconds.count() > idle_timeout_)
+            {
+                RCLCPP_ERROR(get_logger(), "Minimum Jerk Action timed out: an obstacle was in the way and was not moved.");
+                action_rotation_server_->terminate_all();
+                return false;
+            }
+        }
+        return true;
+    }
+    double MinimumJerkRos::compute_angle_()
+    {
+        double angle = abs(this->current_pose_.get_theta() - this->init_pose_.get_theta());
+        angle -= 2 * M_PI * std::floor((angle + M_PI) / (2 * M_PI));
+        angle = abs(angle);
+        return angle;
+    }
+
+    void MinimumJerkRos::stop_rotation_(geometry_msgs::msg::Twist angular_vel)
```

Comment:
you can do a single stop_robot function and use it instead of both stop rotation and stop translation